src/main/handlers/leaves.py

```diff
@@ -0,0 +1,70 @@
+import asyncio
+import random
+
+from datetime import datetime, timedelta
+
+import discord
+
+
+ONGOING = 'ongoing'
+MESSAGE_COUNT = 'message_count'
+NUM_WAIT_MESSAGES = 'number_of_messages_to_wait_for'
+LEAVES_SENT = 'leaves_sent'
+TIMES_REPEAT = 'times_to_repeat'
+LAST_LEAVES = 'last_leaves'
+
+class Leaves:
+
+    def __init__(self, client, config):
+        self.client = client
+        self.config = config
+
+        self.leaves_emoji = '🍃'
+        self.leaves_alias = ':leaves:'      
+        self.leaves_destroyer = ':fire:'
+        self.leaves_sources = ['🍃', '🌳']  
+        self.cooldown = 30
+
+        self.server_vars = {}
+
+    async def execute(self, message):
+        if not message.guild.id in self.server_vars:
+            self.server_vars[message.guild.id] = {}
+            self.server_vars[message.guild.id][ONGOING] = False
+            self.server_vars[message.guild.id][MESSAGE_COUNT] = 0
+            self.server_vars[message.guild.id][NUM_WAIT_MESSAGES] = 5
+            self.server_vars[message.guild.id][LEAVES_SENT] = 0
+            self.server_vars[message.guild.id][TIMES_REPEAT] = 5
+            self.server_vars[message.guild.id][LAST_LEAVES] = datetime.min
+
+        if self.server_vars[message.guild.id][ONGOING]:
```

Comment:
instead of checking if a leaves is happening on every message, why not start a loop on the initial trigger? `for i in range(TIMES_REPEAT):`

Reply:
A loop would most likely create a column of leaves instead of the intended function which is a new leaf after some messages have been sent to appear as if it's falling through the chat.